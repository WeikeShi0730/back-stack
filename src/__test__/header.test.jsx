import { render, renderer } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Header from "../components/header/header.component";

describe("<Header />", () => {
  test("<Header />", () => {
    const { baseElement } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
