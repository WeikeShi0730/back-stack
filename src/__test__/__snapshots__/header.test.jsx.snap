// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> <Header /> 1`] = `
<body>
  <div>
    <nav
      class="sticky top-0 z-50"
    >
      <div
        class="flex h-20 items-center flex-wrap bg-gray-300 bg-opacity-70 backdrop-blur-lg p-3 text-sm md:text-base lg:text-lg"
      >
        <div
          class="flex flex-shrink-0 text-white mx-6"
        >
          <a
            href="/"
          >
            <img
              alt="Back Stack"
              class="fill-current h-10"
              src="/logo.png"
            />
          </a>
        </div>
        <div
          class="hidden lg:flex flex-shrink-0 text-white mx-2"
        >
          <a
            href="/"
          >
            <div
              class="px-4 py-2 leading-none rounded text-black hover:bg-gray-300"
            >
              Home üè†
            </div>
          </a>
        </div>
        <div
          class="hidden lg:flex flex-shrink-0 text-white mx-2"
        >
          <a
            href="/measure"
          >
            <div
              class="px-4 py-2 leading-none rounded text-black hover:bg-gray-300"
            >
              Measure üìê
            </div>
          </a>
        </div>
        <div
          class="hidden lg:flex flex-shrink-0 text-white mx-2"
        >
          <a
            href="/excercise-report"
          >
            <div
              class="px-4 py-2 leading-none rounded text-black hover:bg-gray-300"
            >
              Excercise report üìí
            </div>
          </a>
        </div>
        <div
          class="hidden lg:flex flex-shrink-0 text-white mx-2"
        >
          <a
            href="/device-connection"
          >
            <div
              class="px-4 py-2 leading-none rounded text-black hover:bg-gray-300"
            >
              Device connection üì∂
            </div>
          </a>
        </div>
        <div
          class="lg:flex flex-shrink-0 flex-grow"
        />
        <div
          class="hidden flex-shrink-0 lg:flex lg:items-center lg:w-auto"
        >
          <a
            href="/login"
          >
            <button
              class="px-4 py-2 leading-none rounded text-black hover:bg-gray-300 lg:mt-0 font-light"
            >
              Sign in
            </button>
          </a>
        </div>
        <div
          class="flex items-center justify-between lg:hidden mx-5"
        >
          <button
            class="flex h-6 w-6 items-center space-x-2 focus:outline-none font-light"
          >
            <div
              class="w-6 flex items-center justify-center relative"
            >
              <span
                class="hamburger -translate-y-2"
              />
              <span
                class="hamburger opacity-100"
              />
              <span
                class="hamburger translate-y-2"
              />
            </div>
          </button>
        </div>
      </div>
      <div
        class="false"
      />
      <div
        class="absolute h-screen top-0 right-0 -z-10 pt-20 text-sm md:text-base lg:text-lg"
      />
    </nav>
  </div>
</body>
`;
